components/ModEditFacilitySection.vue

```
@@ -266,4 +270,38 @@ const listPrefectureJapanJa: Ref<string[]> = ref([
     '石川県', '福井県', '山梨県', '長野県', '岐阜県', '静岡県', '愛知県', '三重県', '滋賀県', '京都府', '大阪府',
     '兵庫県', '奈良県', '和歌山県', '鳥取県', '島根県', '岡山県', '広島県', '山口県', '徳島県', '香川県', '愛媛県',
     '高知県', '福岡県', '佐賀県', '長崎県', '熊本県', '大分県', '宮崎県', '鹿児島県', '沖縄県'])
+
+onBeforeMount(async () => {
+    isFacilitySectionInitialized.value = false
+
+    const route = useRoute()
+
+    // Wait for the route to be fully resolved
+    await nextTick()
+
+    if (!facilityStore.facilityData.length) {
+        await facilityStore.getFacilities()
+    }
+
+    // Ensure the route param `id` is available before proceeding
+    const id = route.params.id
+    if (id) {
```

should throw  an error here if it's not available. I'd do an inverted if.
```
if(!id) {
    console.error(...)
   toast()
return
}
```